### PR TITLE
Add D-Bus support

### PR DESCRIPTION
**Pull request by MRZA-MRZA**
Some initial D-Bus support. It constantly reports current hash rate using D-Bus so it can be caught by external application (for monitoring purposes). To compile with D-Bus support add "-DETHDBUS=ON" to cmake command. D-Bus interface is "eth.miner.monitor". Signal name is "Hash".

### CODE REVIEW

File: ethminer/MinerAux.h [hunk 1]
```diff
@@ -774,8 +777,12 @@ class MinerCLI
 				{
 					auto mp = f.miningProgress();
 					f.resetMiningProgress();
-					if (current)
+					if (current){
```
Comment (davidf2281) [inline ethminer/MinerAux.h]: Existing style throughout this project would appear to be opening braces on new line.

File: ethminer/MinerAux.h [hunk 2]
```diff
@@ -778,6 +781,9 @@ class MinerCLI
 					{
 						minelog << "Mining on" << current.header << f.getSolutionStats();
 						minelog << mp;
+#if ETH_DBUS
+						dbusint.send(toString(mp).c_str());
```
Comment (chfast) [inline ethminer/MinerAux.h]: In C++11 you can use `.data()` instead of `.c_str()`.

File: ethminer/DBusInt.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+{
+public:
+		DBusInt()
+		{
+			dbus_error_init(&err);
+			conn = dbus_bus_get(DBUS_BUS_SESSION, &err);
+			if (!conn)
+			{
+				cerr << "DBus error " << err.name << ": " << err.message << endl;
+				exit(-1);
```
Comment (chfast) [inline ethminer/DBusInt.h]: Please do not exit of D-bus error.

File: ethminer/DBusInt.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+			}
+			cout << "DBus initialized!" << endl;
+		}
+
+		void send(const char *hash)
+		{
+			DBusMessage *msg;
+			msg = dbus_message_new_signal("/eth/miner/hash", "eth.miner.monitor", "Hash");
+			if (msg == NULL)
```
Comment (chfast) [inline ethminer/DBusInt.h]: ~~NULL~~nullptr

**Comment by chfast** [comment]
Can you also create a `CODEOWNERS` file and add yourself there for D-bus?
See https://github.com/blog/2392-introducing-code-owners.

**Comment by MRZA-MRZA** [comment]
Sorry, I'm not familiar with CODEOWNERS. I added it to the root of the repository.

**Comment by chfast** [comment]
Me neither, but I wanted to give it a try.

**Comment by aliceaia** [comment]
-- Checking for module 'dbus-1'
--   No package 'dbus-1' found

**Comment by MRZA-MRZA** [comment]
@aliceaia what distributive?

**Comment by aliceaia** [comment]
@MRZA-MRZA 
I use ubuntu

**Comment by aliceaia** [comment]
@MRZA-MRZA 
Sorry for my bad english, maybe I misunderstood you.

**Comment by MRZA-MRZA** [comment]
@aliceaia which version? Please show output of pkg-config --modversion dbus-1

**Comment by aliceaia** [comment]
m1@m1-desktop:~/debian$ pkg-config --modversion dbus-1
1.10.10

**Comment by aliceaia** [comment]
@MRZA-MRZA 
I will try reinstalling the operating system

**Comment by MRZA-MRZA** [comment]
@aliceaia You don't need to. Does this file exist? /usr/lib/x86_64-linux-gnu/pkgconfig/dbus-1.pc

**Comment by aliceaia** [comment]
@MRZA-MRZA 
No, I checked and did not have that file

**Comment by MRZA-MRZA** [comment]
@aliceaia please show output of
pkg-config --variable pc_path pkg-config

**Comment by aliceaia** [comment]
@MRZA-MRZA 
I have reinstalled the operating system and have successfully build. Thank you very much.

**Comment by aliceaia** [comment]
Now I need an application to connect to ethminer via dbus right?

**Comment by MRZA-MRZA** [comment]
@aliceaia 
```
#!/usr/bin/env python

import dbus
import dbus.mainloop.glib
import gobject

def catch_signal_handler(hash_string):
	print hash_string

if __name__ == '__main__':
	dbus.mainloop.glib.DBusGMainLoop(set_as_default=True)
	bus = dbus.SessionBus()
	bus.add_signal_receiver(catch_signal_handler, dbus_interface = "eth.miner.monitor", signal_name = "Hash")
	loop = gobject.MainLoop()
	loop.run()
```
You can use this sample python code

**Comment by trungtt1981** [comment]
I always get this compiler error when building on Ubuntu

```
ethminer/ethminer/DBusInt.h:1:23: fatal error: dbus/dbus.h: No such file or directory
compilation terminated.

```

Although, I have dbus installed

```
pkg-config --modversion dbus-1
1.10.6

```

```
ls /usr/lib/x86_64-linux-gnu/pkgconfig/dbus-1.pc
-rw-r--r-- 1 root 623 Jan 12  2017 /usr/lib/x86_64-linux-gnu/pkgconfig/dbus-1.pc
```
BuildInfo

```
#define ETH_PROJECT_VERSION "0.13.0.dev0"
#define ETH_BUILD_TYPE "Release"
#define ETH_BUILD_PLATFORM "Linux/g++"

```

**Comment by chfast** [comment]
You probably need dbug dev package installed.

**Pull request merged**